libsplinter/src/registry/diesel/operations/add_node.rs

```diff
@@ -37,11 +36,6 @@ pub(in crate::registry::diesel) trait RegistryAddNodeOperation {
 #[cfg(feature = "postgres")]
 impl<'a> RegistryAddNodeOperation for RegistryOperations<'a, diesel::pg::PgConnection> {
     fn add_node(&self, node: Node) -> Result<(), RegistryError> {
-        // Verify that the node's required fields are non-empty
```

Comment:
This change will require making the fields of the `Node` struct private; otherwise you could construct a `Node` directly without the builder that would violate these constraints.

Reply:
Added a new commit to change the fields to private